samples/matter/thermostat/src/temperature_measurement/sensor.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "sensor.h"
+#include "temp_sensor_manager.h"
+
+#include <stdint.h>
+
+#include <temp_sensor_manager.h>
+
+#include <DeviceInfoProviderImpl.h>
+
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <controller/ReadInteraction.h>
+#include <zephyr/logging/log.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceLayer;
+using namespace ::chip::app;
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+namespace
+{
+
+constexpr chip::EndpointId mTemperatureMeasurementEndpointId = 1;
+constexpr uint16_t startingMockedValue = 1600;
+
+constexpr uint16_t kSimulatedReadingFrequency = (60000 / kSensorTimerPeriodMs); /*Change simulated number every minute*/
+
+#if CONFIG_THERMOSTAT_TEMPERATURE_STEP == 0
+constexpr int16_t sMockTemp[] = { 2000, 2731, 1600, 2100, 1937, 3011, 1500, 1899 };
+#endif
+
+} // namespace
+
+TemperatureSensor::TemperatureSensor()
+{
+	mMockTempIndex = 0;
+	mPreviousTemperature = startingMockedValue;
+	BindingHandler::Init();
+}
+
+void TemperatureSensor::FullMeasurement()
+{
+	InternalMeasurement();
+	ExternalMeasurement();
+}
+
+void TemperatureSensor::InternalMeasurement()
+{
+	uint16_t temperature;
+#if CONFIG_THERMOSTAT_TEMPERATURE_STEP == 0
+
+	temperature = sMockTemp[mMockTempIndex];
+
+	mCycleCounter++;
+	if (mCycleCounter >= kSimulatedReadingFrequency) {
+		if (mMockTempIndex >= ArraySize(sMockTemp) - 1) {
+			mMockTempIndex = 0;
+		} else {
+			mMockTempIndex++;
+		}
+		mCycleCounter = 0;
+	}
+#else
+	uint16_t nextValue = mPreviousTemperature + CONFIG_THERMOSTAT_TEMPERATURE_STEP;
```

Comment:
```suggestion
	int16_t nextValue = mPreviousTemperature + CONFIG_THERMOSTAT_TEMPERATURE_STEP;
```
so that it doesn't underflow so easily when the step is negative. Then you will have to add `static_cast<uint16_t>(nextValue)`.